### PR TITLE
[4.0] Fix js ajax for pre update checker

### PR DESCRIPTION
**Pull request by degobbis**
### Summary of Changes
Fix js ajax for pre update checker

### Testing Instructions
- Install an extension
- Go to joomla update
[update 07.07.2020]
- Click on `Options` on the right corner
- Change `Update Chanel` to `Custom Url`
- Change `Minimum Stability` to `Development`
- Insert `https://update.joomla.org/core/nightlies/next_major_list.xml` as `Custom URL` than `Save and close`
- Check for updates

### Actual result BEFORE applying this Pull Request
Notice the pre update checker cant find the combat tag
All installed extensions show the flag: Missing Compatibility Tag

grafik

### Expected result AFTER applying this Pull Request
The pre update checker can now find the combat tag (npm required)
All installed extensions without update server show the flag: `Missing Compatibility Tag`
All installed extensions with update server but without update block with a `<targetplatform>` tag for Joomla 4 show the flag: `No`
All installed extensions with update server and an update block with a `<targetplatform>` tag for Joomla 4 show the flag: `Yes`

grafik

### CODE REVIEW

File: build/media_source/com_joomlaupdate/js/default.es5.js [hunk 1]
```diff
@@ -151,17 +151,13 @@ Joomla = window.Joomla || {};
 
 		// Request the server to check the compatiblity for the passed extension and joomla version
 		Joomla.request({
-			url: PreUpdateChecker.config.serverUrl,
-			data: {
-				'joomla-target-version': PreUpdateChecker.joomlaTargetVersion,
-				'extension-id': node.getAttribute('data-extensionId')
-			},
+			url: PreUpdateChecker.config.serverUrl + '&' + 'joomla-target-version=' + PreUpdateChecker.joomlaTargetVersion + '&' + 'extension-id=' + node.getAttribute('data-extension-id'),
```
Comment (Fedik) [inline build/media_source/com_joomlaupdate/js/default.es5.js]: please use `encodeURIComponent()` for URL parameters, eg:
 `encodeURIComponent(PreUpdateChecker.joomlaTargetVersion)`

Comment (degobbis) [reply build/media_source/com_joomlaupdate/js/default.es5.js]: Thx for the hint

**Comment by degobbis** [comment]
I have updated the test instructions because the Pre-Checker was not triggered in the default configuration.

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on f0f1cd1eb29961118ed6b61027bc8978e23fa5d5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29980">issues.joomla.org/tracker/joomla-cms/29980</a>.</sub>

**Comment by chmst** [comment]
screen shot 2020-07-07 at 20 05 28

Result after patch<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29980">issues.joomla.org/tracker/joomla-cms/29980</a>.</sub>

**Comment by bonzani** [comment]
I have tested this item :white_check_mark: successfully on d876f4c4b727efd337c2e17c4e4e74c059bbe739<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29980">issues.joomla.org/tracker/joomla-cms/29980</a>.</sub>

**Comment by degobbis** [comment]
RTC

**Comment by SharkyKZ** [comment]
RTC.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29980">issues.joomla.org/tracker/joomla-cms/29980</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**